rule.go

```diff
@@ -419,6 +419,21 @@ func (p PCRE) String() string {
 	return s.String()
 }
 
+//string returns a string for all of the flowbits
+func (fb Flowbit) String() string {
+	var s strings.Builder
+	if inSlice(fb.Action, []string{"noalert", "isset", "isnotset", "set", "unset", "toggle"}) {
```

Comment:
In order to reduce indentation you should use guard clauses[0], something like this:

```
if !inSlice(fb.Action, []string{"noalert", "isset", ...}) {
  return ""
}
var s strings.Builder
s.WriteString(fmt.Sprintf("flowbits:%s", fb.Action))
if fb.Value != "" {
  s.WriteString(fmt.Sprintf(",%s", fb.Value))
}
s.WriteString(";")
return s.String()
```  
**

[0] https://medium.com/@scadge/if-statements-design-guard-clauses-might-be-all-you-need-67219a1a981a
** note that I didn't test this code. Please verify.